tests/functional/idempotency/test_idempotency.py

```diff
@@ -274,6 +274,40 @@ def lambda_handler(event, context):
     stubber.assert_no_pending_responses()
     stubber.deactivate()
 
+@pytest.mark.parametrize("idempotency_config", [{"use_local_cache": False}, {"use_local_cache": True}], indirect=True)
+def test_idempotent_lambda_first_execution_event_mutation(
```

Comment:
This test passes without this change, as the `expected_params_update_item` uses the exact same hash `expected_params_put_item` uses in the first operation -- unsure yet whether that's an issue on the test setup.

I'm gonna launch a fresh stack with the latest code to isolate it

Reply:
yep, it's the test setup. Repro code: https://github.com/heitorlessa/pt-1090

**Pass by reference (latest code)**

![image](https://user-images.githubusercontent.com/3340292/161928707-8953fd87-f01d-4045-916d-e6cf188b8a03.png)

**Pass by value (with fix)**

![image](https://user-images.githubusercontent.com/3340292/161928844-747fc8a4-f115-4c87-b0ad-8547626e5b23.png)